sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -892,4 +892,150 @@ public void testUpdateByIdWithSubQuery() throws Exception {
         assertThat(printedTable(response.rows()), is("1| Slartibartfast\n" +
                                                      "2| Trillian\n"));
     }
+
+    @Test
+    public void test_update_by_id_returning_id() throws Exception {
+        execute("create table test (id int primary key, message string) clustered into 2 shards");
+        execute("insert into test values(1, 'msg');");
+        assertEquals(1, response.rowCount());
+        refresh();
+
+        execute("update test set message='updated' where id = 1 returning id");
+
+        assertThat(printedTable(response.rows()), is("1\n" ));
+        refresh();
+    }
+
+    @Test
+    public void test_update_by_id_returning_function() throws Exception {
+        execute("create table test (id int primary key, message string) clustered into 2 shards");
+        execute("insert into test values(1, 'msg');");
+        assertEquals(1, response.rowCount());
+        refresh();
+
+        execute("update test set message='updated' where id = 1 returning id + 1");
+
+        assertThat(printedTable(response.rows()), is("2\n" ));
+        refresh();
+    }
+
+    @Test
+    public void test_update_by_id_returning_id_with_outputname() throws Exception {
+        execute("create table test (id int primary key, message string) clustered into 2 shards");
+        execute("insert into test values(1, 'msg');");
+        assertEquals(1, response.rowCount());
+        refresh();
+
+        execute("update test set message='msg' where id = 1 returning id as renamed");
+
+        assertThat((response.cols()[0]), is("renamed" ));
+        refresh();
+    }
+
+    @Test
+    public void test_update_by_id_returning_function_with_outputname() throws Exception {
+        execute("create table test (id int primary key, message string) clustered into 2 shards");
+        execute("insert into test values(1, 'msg');");
+        assertEquals(1, response.rowCount());
+        refresh();
+
+        execute("update test set message='updated' where id = 1 returning id + 1 as foo");
+
+        assertThat((response.cols()[0]), is("foo" ));
+        assertThat(response.rows()[0][0], is(2));
+
+        refresh();
+    }
+
+    @Test
+    public void test_update_by_id_returning_multiple_fields() throws Exception {
```

Comment:
Could some of the tests be converted to unit tests, or do all of these have to be on a integration test level?